hudi-common/src/main/java/org/apache/hudi/common/table/read/FileGroupRecordBuffer.java

```diff
@@ -455,8 +526,10 @@ protected Pair<Boolean, T> merge(BufferedRecord<T> olderRecord, BufferedRecord<T
           Comparable oldOrderingValue = olderRecord.getOrderingValue();
           if (!olderRecord.isCommitTimeOrderingDelete()
               && oldOrderingValue.compareTo(newOrderingValue) > 0) {
+            updatePartiallyIfNeeded(olderRecord, newerRecord, readerSchema, readerSchema, partialUpdateMode);
```

Comment:
if not for partial updates, we end up picking entire old record as is. 
w/ partial updates, I see here we are switching the ordering. but isn't the expectation from `updatePartiallyIfNeeded` method is that, first argument is likely fully formed record and 2nd argument contains only partial update and the method merges both records into newerRecord. 

but if we switch the ordering, does the method impl know that newRecord is partial and old record is fully formed and act accordingly ?